Code/MinimalLib/minilib.cpp

```diff
@@ -908,3 +908,22 @@ JSLog *set_log_capture(const std::string &log_name) {
 void enable_logging() { RDKit::MinimalLib::LogHandle::enableLogging(); }
 
 void disable_logging() { RDKit::MinimalLib::LogHandle::disableLogging(); }
+
+#ifdef RDK_BUILD_MINIMAL_LIB_FRAGMENTATION
+std::pair<std::vector<std::string>, std::vector<std::string>> fragmentMol(const JSMol &mol, unsigned int minCuts, unsigned int maxCuts, unsigned int maxCutBonds) {
+
+  std::vector<std::pair<ROMOL_SPTR, ROMOL_SPTR>> res;
+  RDKit::MMPA::fragmentMol(*(mol.d_mol.get()), res, minCuts, maxCuts, maxCutBonds);
+
+  size_t length = res.size();
+  std::vector<std::string> fragments1(res.size());
+  std::vector<std::string> fragments2(res.size());
+
+  for (size_t res_idx = 0; res_idx < res.size(); res_idx++) {
+    fragments1[res_idx] = (res[res_idx].first.get() ? (MolToSmiles(*res[res_idx].first, true)) : "");
```

Comment:
Is it correct that res[res_idx].first contains a pointer? and get() method return a value by this pointer? Did you tried on malformed data (to amke sure that in case mol could not be created first.get() returns null)?